src/jshint.js

```diff
@@ -653,12 +653,15 @@ var JSHINT = (function() {
           case "3":
           case "5":
           case "6":
+          case "7":
```

Comment:
I like this idea, but the variability in warning messages may fragment Google/StackOverflow search results. What do you think about extending the message to suggest both alternatives, e.g.

> 'Exponentiation operator' is only available in ES7 (use 'esversion: 7' or 'esversion: 2016').

...or (maybe better) omitting the suggested value altogether:

> 'Exponentiation operator' is only available in ES7. Enable it with the 'esversion' option.